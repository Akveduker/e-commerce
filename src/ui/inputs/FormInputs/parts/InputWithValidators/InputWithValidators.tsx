import { FC } from 'react';
import DefaultInput from '../../../DefaultInput/DefaultInput';
import InputName from '../InputName/InputName';
import InputValidationError from '../InputValidationError/InputValidationError';
import InputValidationOk from '../InputValidationOk/InputValidationOk';
import s from './InputWithValidators.module.scss'
import { InputWithValidatorsProps } from '../../../../../types/formInput/formInput';
const InputWithValidators: FC<InputWithValidatorsProps> = ({ inputTitle, errorBody, errorTitle, status, ...props }) => {
    return (
        <label>
            <InputName inputName={inputTitle} />
            <DefaultInput labelClassName={s['container']}  {...props} >
                <InputValidationOk
                    status={status}
                />
            </DefaultInput>
            <InputValidationError
                status={status}
                errorBody={`${errorTitle} ${errorBody}`}
            />
        </label>
    );
};

export default InputWithValidators;