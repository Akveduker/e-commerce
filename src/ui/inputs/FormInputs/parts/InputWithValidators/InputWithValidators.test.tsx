import { render, screen } from "@testing-library/react"
import InputWithValidators from "./InputWithValidators"

describe('InputWithValidators', () => {
    test('Все элементы рендерятся правильно', () => {
        const { container } = render(
            <InputWithValidators
                inputTitle="title"
                status="succeeded"
                errorBody="error"
                errorTitle="error"
            />
        )
        expect(screen.queryByText('title')).toBeInTheDocument()
        expect(screen.queryByTestId('icon')).toBeInTheDocument()
        expect(screen.queryByText('error')).toBeNull()
        expect(container).toMatchSnapshot()
    })
})