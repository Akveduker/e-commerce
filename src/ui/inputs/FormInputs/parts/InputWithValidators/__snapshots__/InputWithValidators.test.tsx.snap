// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithValidators Все элементы рендерятся правильно 1`] = `
<div>
  <label>
    <p
      class="name"
    >
      title
    </p>
    <label
      class="with-children container"
      data-testid="label"
    >
      <input
        class="with-children__input"
        data-testid="default"
      />
      <div
        class="with-children__children"
      >
        <div>
          <div
            class="icon"
            data-testid="icon"
          >
            <svg
              fill="none"
              height="25"
              viewBox="0 0 25 25"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.2495 24.1758C18.8769 24.1758 24.2495 18.8032 24.2495 12.1758C24.2495 5.54836 18.8769 0.175781 12.2495 0.175781C5.6221 0.175781 0.249512 5.54836 0.249512 12.1758C0.249512 18.8032 5.6221 24.1758 12.2495 24.1758Z"
                fill="#6A983C"
              />
              <path
                d="M18.3068 6.8042L10.5354 14.5756L7.33535 11.3756L5.73535 12.9756L10.5354 17.7756L19.9068 8.4042L18.3068 6.8042Z"
                fill="#F4F8EC"
              />
            </svg>
          </div>
        </div>
      </div>
    </label>
    <div />
  </label>
</div>
`;
